### PR TITLE
Introduce ITCSS Structure in the (S)CSS Code

### PR DESCRIPTION
**Pull request by nachtfunke**
Hello!

I saw that while the project uses SCSS, it doesn't utilise it. I saw the potential to add a bit of methodology in order to add a layer of organisation to the SCSS Code. This follows _basically_ the ITCSS inverted triangle of specificity, with the only difference that `components` is named `chunks`, because we _usually_ nowadays don't think of CSS code as the source of truth for components.

- Splits the code according to ITCSS into separate partials. Partials in SCSS won't be rendered by the compiler.
- Utilises the sass module syntax via `@use`, including index files for objects and chunks: `styles/objects/_index.scss`
- Moves breakpoints into a `$breakpoints` map in `_settings.scss`, retrievable via a `breakpoint()` function, in `_tools.scss`
- Additionally, I split the `@font-face` rulesets into their own file, to implement a common pattern intended to make the site a _bit_ more performant, accepting a short FOUT in the case of uncached font files. This also eliminates them as a render-blocking resource.

***

_Please_ (!) feel free to reject this. I believe this might be valuable going forward _but_ it might not be!

### CODE REVIEW

**Review by Holben888 — COMMENTED** [review]
I _love_ bringing some organization to this SCSmess! I do have one criticism though: I agree that "components" is kinda tough to justify as a concept these days, but "chunks" really blurs the line with "objects" for me. Would you be open to merging objects and chunks into a single chunks directory? This still keeps the major pieces of the triangle around, but avoids some confusion for future contributors that may or may not know ITCSS.

File: docs/src/_includes/base-layout.html [hunk 1]
```diff
@@ -5,6 +5,7 @@
   <meta charset="UTF-8">
   <meta http-equiv="X-UA-Compatible" content="IE=edge">
   <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <link rel="stylesheet" href="/styles/fonts.scss" media="print" onload="this.media='all'">
```
[the remark this answers is not in the document]
Comment (nachtfunke) [reply docs/src/_includes/base-layout.html]: I've added the ressource hints for the fonts! The `media="print" onload="this.media='all'"` trick will tell the browser to treat _this_ stylesheet as non-render blocking.

[the remark this answers is not in the document]
Comment (nachtfunke) [reply docs/src/_includes/base-layout.html]: pro forma, I've also merged the two directories!

**Review by Holben888 — APPROVED** [review]
Love this, thanks! I'll be sure to add some notes on ITCSS to our (coming) `CONTRIBUTING` doc too. Looks like you'll need to rebase onto that tiny fix to our docs though 👍

**Comment by nachtfunke** [comment]
> Love this, thanks! I'll be sure to add some notes on ITCSS to our (coming) `CONTRIBUTING` doc too. Looks like you'll need to rebase onto that tiny fix to our docs though 👍

+ Rebase `main` onto this branch done.

**Pull request merged**